solidity-plugin/src/main/java/org/sonarsource/solidity/MetricsVisitor.java

```diff
@@ -0,0 +1,189 @@
+  // .stream()
+  // .map(x -> x.functionDefinition())
+  // .filter(Objects::nonNull)
+  // .collect(Collectors.toList());
+  //
+  // fileMeasures.setFunctionNumber(functionList.size());
+  //
+  // long statementsCounter = 0;
+  // List<StatementContext> statementList = new ArrayList<>();
+  //
+  // functionList
+  // .stream()
+  // .map(FunctionDefinitionContext::block)
+  // .map(BlockContext::statement)
+  // .filter(Objects::nonNull)
+  // .forEach(x -> statementList.addAll(x));
+  //
+  // BlockVisitorClass vis = new BlockVisitorClass();
+  // functionList
+  // .stream()
+  // .map(x -> x.accept(vis))
+  // .forEach(x -> System.out.println(x));
+  // return super.visitSourceUnit(ctx);
+  // }
+
+  @Override
+  public Token visitContractDefinition(ContractDefinitionContext ctx) {
+    contractCounter++;
+    return super.visitContractDefinition(ctx);
+  }
+
+  @Override
+  public Token visitContractPart(ContractPartContext ctx) {
+    return super.visitContractPart(ctx);
+  }
+
+  @Override
+  public Token visitFunctionDefinition(FunctionDefinitionContext ctx) {
+    functionCounter++;
+    return super.visitFunctionDefinition(ctx);
+  }
+
+  @Override
+  public Token visitBreakStatement(BreakStatementContext ctx) {
+    statements++;
+    return super.visitBreakStatement(ctx);
+  }
+
+  @Override
+  public Token visitContinueStatement(ContinueStatementContext ctx) {
+    statements++;
+    return super.visitContinueStatement(ctx);
+  }
+
+  @Override
+  public Token visitDoWhileStatement(DoWhileStatementContext ctx) {
+    statements++;
+    return super.visitDoWhileStatement(ctx);
+  }
+
+  @Override
+  public Token visitEmitStatement(EmitStatementContext ctx) {
+    statements++;
+    return super.visitEmitStatement(ctx);
+  }
+
+  @Override
+  public Token visitReturnStatement(ReturnStatementContext ctx) {
+    statements++;
+    return super.visitReturnStatement(ctx);
+  }
+
+  @Override
+  public Token visitExpressionStatement(ExpressionStatementContext ctx) {
+    statements++;
+    return super.visitExpressionStatement(ctx);
+  }
+
+  @Override
+  public Token visitForStatement(ForStatementContext ctx) {
+    statements++;
+    return super.visitForStatement(ctx);
+  }
+
+  @Override
+  public Token visitIfStatement(IfStatementContext ctx) {
+    statements++;
+    return super.visitIfStatement(ctx);
+  }
+
+  @Override
+  public Token visitThrowStatement(ThrowStatementContext ctx) {
+    statements++;
+    return super.visitThrowStatement(ctx);
+  }
+
+  @Override
+  public Token visitVariableDeclarationStatement(VariableDeclarationStatementContext ctx) {
+    statements++;
+    return super.visitVariableDeclarationStatement(ctx);
+  }
+
+  @Override
+  public Token visitWhileStatement(WhileStatementContext ctx) {
+    statements++;
+    return super.visitWhileStatement(ctx);
+  }
+
+  /*
+   * private static class BlockVisitorClass extends SolidityBaseVisitor {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: This block of commented-out lines of code should be removed. ([squid:CommentedOutCodeLine](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3ACommentedOutCodeLine))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=16&issues=AWQTGK7hbXXTsx7O4tIN&open=AWQTGK7hbXXTsx7O4tIN)

[ISSUE_KEY]: AWQTGK7hbXXTsx7O4tIN